File: c++/leetcode/338. Counting Bits.cpp
class Solution {
public:  
    vector<int> countBits(int num) {
        vector<int> ans;
        
        const auto count_bits = [](int num) {
            int count{};
            
            while(num && ++count) 
                num &= num - 1;
            
            return count;
        };
        
        for(int i = 0; i <= num; ++i) 
            ans.push_back(count_bits(i));
        
        return ans;
    }
}; 

File: javascript/leetcode/338. Counting Bits.js
/**
 * @param {number} num
 * @return {number[]}
 */
var countBits = function(num) {
    const ans = []
    
    const count_bits = (num) => {
        let count = 0
        
        while(num && ++count)
            num &= num - 1;
        
        return count;
    }
    
    for(let i = 0; i <= num; ++i)
        ans.push(count_bits(i))
    
    return ans;
};
